pandas/_libs/reduction.pyx

```diff
@@ -491,3 +505,83 @@ cdef class BlockSlider:
         mgr.blocks = self.blocks
         mgr._blklocs = self.orig_blklocs
         mgr._blknos = self.orig_blknos
+
+
+cdef class ArraySlider(FrameSlider):
```

Comment:
i doubt it will be big, but there might be some speedups available py doing `@cython.final` on both of BlockSlider and ArraySlider